src/blobs.jl

```diff
@@ -72,6 +72,16 @@ end
 delete(x::Container, key::String; kw...) = Azure.delete(API.makeURL(x, key); kw...)
 delete(x::Object; kw...) = delete(x.store, x.key; credentials=x.credentials, kw...)
 
+function create_append_blob(c::Container, key::String; kw...)
+    Azure.put(API.makeURL(c, key), ["x-ms-blob-type" => "AppendBlob"], ""; kw...)
+end
+
+# https://learn.microsoft.com/en-us/rest/api/storageservices/append-block
+function append_block(c::Container, key::String, data::AbstractVector{UInt8}; kw...)
+    url = API.makeURL(c, key)
+    Azure.put(url, [], data; query=Dict("comp" => "appendblock"), kw...)
```

Comment:
So ideally, we'd utilize the parallel multipart upload for larger files, as we do in `API.putObjectImpl`. This would need some tweaking because currently, `API.uploadPart` sets `"comp" => "block"` (couple of lines above in this file), so we'd probably want to propagate the query to it.

Reply:
But if you need to append every couple of seconds or minutes say 32MiB chunks of data at a time, then you'd still benefit from multipart uploads, no?

Reply:
I suppose there could be big spikes in the log stream rate, but in my use case I haven't seen spikes big enough to necessitate uploading in parallel.

Looking at the Azure docs, I'm not sure append blobs support parallel appending: https://learn.microsoft.com/en-us/rest/api/storageservices/append-block?tabs=azure-ad#remarks As far as I can tell, once you make an append blob you can only append one at a time. Otherwise there'd have to be some scheme to tell it what block id you're appending; I don't see that in these docs.